code/modules/mob/living/living_defines.dm

```diff
@@ -58,3 +58,6 @@
 	var/unique_name = 0 //if a mob's name should be appended with an id when created e.g. Mob (666)
 
 	var/list/butcher_results = null
+
+	var/stun_absorption = FALSE //If all incoming stuns are being absorbed

```

Comment:
I put it right into `living_defines.dm` as a framework for other stun immunity effects that could be implemented in the future. If you have a better way of doing this, I'd be happy to oblige you.